src/Layer/OrientedImageLayer.js

```diff
@@ -0,0 +1,169 @@
+        layer.background.material = layer.material || layer.background.material;
+    }
+}
+
+function createBackground(radius) {
+    if (!radius || radius <= 0) { return undefined; }
+    var geometry = new THREE.SphereGeometry(radius, 32, 32);
+    var material = new THREE.MeshPhongMaterial({
+        color: 0x7777ff,
+        side: THREE.DoubleSide,
+        transparent: true,
+        opacity: 0.5,
+        wireframe: true,
+    });
+    var sphere = new THREE.Mesh(geometry, material);
+    sphere.visible = true;
+    sphere.name = 'OrientedImageBackground';
+    return sphere;
+}
+
+/**
+ * @classdesc OrientedImageLayer loads oriented images, and project these textures on the scene.
+ * It is design to create an immersive view. </br>
+ * It loads a set of panoramic position and orientation,
+ * a set of camera calibration file (it's the same set of camera for each panoramic),
+ *  and a set of texture (each image for each camera for each panoramic), all organised in an {@link OrientedImageSource}. </br>
+ * It creates an {@link OrientedImageMaterial} used to do projective texture mapping on the scene.
+ * @extends GeometryLayer
+ */
+class OrientedImageLayer extends GeometryLayer {
+    /**
+     * @constructor
+     * @param { string } id - Configuration object
+     * @param { Object } config - configuration of the layer
+     * @param { OrientedImageSource } config.source - Source of the datas
+     */
+    constructor(id, config = {}) {
+        super(id, new THREE.Group(), config);
+
+        this.background = config.background || createBackground(config.sphereRadius);
+
+        this.object3d.add(this.background);
+
+        // currentPano is the current point, means it's the closest from the camera
+        this.currentPano = undefined;
+
+        // store a callback to fire event when current panoramic change
+        this.onPanoChanged = config.onPanoChanged || (() => {});
+
+        // panos is an array of feature point, representing many panoramics.
+        // for each point, there is a position and a quaternion attribute.
+        const p1 = GeoJsonParser.parse(config.orientation, {
+            mergeFeatures: false,
+            crsOut: config.projection }).then((res) =>  {
+            this.panos = res.features;
+
+            // we need to know if we are in globe view, for orientation.
+            const isGlobe = config.projection == 'EPSG:4978';
+
+            // add position and quaternion attributes from point feature
+            let i = 0;
+            for (const pano of this.panos) {
+                // set position
+                coord.set(pano.crs, pano.vertices[0], pano.vertices[1], pano.vertices[2]);
+                pano.position = coord.xyz();
+                // set quaternion
+                pano.quaternion = OrientationUtils.quaternionFromAttitude(pano.geometry[0].properties, coord, isGlobe);
+
+                // TODO clean DataSourceProvider, so that we don't have this hack to do
```

Comment:
the hack is only  `pano.material = {};` ?

Reply:
yes, it's done here #990

Reply:
good